### PR TITLE
add new contract for bulk migrate 

### PR DESCRIPTION
**Pull request by 0xaptosj**
Note: names must be passed in in the correct order (e.g. domain_names: [d1, d1], subdomain_names: [none, s1]), otherwise tx will fail (and no name will be migrated)

also add a publish script for testing

### CODE REVIEW

File: bulk_migrate/README.md [hunk 1]
```diff
@@ -1,9 +1,6 @@
 This script helps with bulk name migration
 
-Preparation:
-1. Update `Move.toml` with the correct `repository` account.
-
 Each iteration:
 1. Update the script with the appropriate names to migrate
 2. Compile the script: `aptos move compile`
-3. Run the script: `aptos move run-script --compiled-script-path bulk_migrate/build/bulk_migrate/bytecode_scripts/main.mv --profile admin`
+3. Run the script: `aptos move run-script --compiled-script-path bulk_migrate/build/bulk_migrate/bytecode_scripts/main.mv --profile admin bulk_migrate=address_of_user_initiating_migration`
```
Comment (0xaptosj) [inline bulk_migrate/README.md:6]: what should be address of `bulk_migrate`? should it be the address that deploys the bulk migrate contract to?

File: bulk_migrate/sources/migrate_tests.move [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            0,
+            false,
+            option::none(),
+            option::none(),
+        );
+
+        // Bump mode
+        router::set_mode(router, 1);
+
+        bulk_migrate::migrate::bulk_migrate_name(
```
Comment (0xaptosj) [inline bulk_migrate/sources/migrate_tests.move:55]: `bulk_migrate::migrate::bulk_migrate_name` please suggest better name

File: bulk_migrate/Move.toml [hunk 3]
```diff
@@ -2,9 +2,6 @@
 name = 'bulk_migrate'
 version = '1.0.0'
 
-[addresses]
```
[the remark this answers is not in the document]
Comment (0xaptosj) [reply bulk_migrate/Move.toml:5]: added back

[the remark this answers is not in the document]
Comment (0xaptosj) [reply bulk_migrate/Move.toml:5]: actually i found we don't specify `addresses` in other module projects (router, core, etc), we only specify in script projects (bulk clear, etc)

[the remark this answers is not in the document]
Comment (0xaptosj) [reply bulk_migrate/Move.toml:5]: after i add back the addresses, i got this when running the test
```
  "Error": "Unexpected error: Unable to resolve packages for package 'bulk_migrate'"
```

File: bulk_migrate/sources/migrate.move [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    public entry fun bulk_migrate_name(
+        user: &signer,
+        domain_names: vector<String>,
+        subdomain_names: vector<Option<String>>,
+    ) {
+        assert!(
+            vector::length(&domain_names) == vector::length(&subdomain_names),
+            error::invalid_argument(EDOMAIN_AND_SUBDOMAIN_MUST_HAVE_SAME_LENGTH)
+        );
+        // Need to reverse because we can only traverse backward
```
[the remark this answers is not in the document]
Comment (0xaptosj) [reply bulk_migrate/sources/migrate.move]: i don't know what's the method to get value by index, looked through the vector lib and didn't find any
```
   │
22 │             let domain_name = domain_names[idx];
   │                               ^^^^^^^^^^^^^^^^^ `_[_]` index operator only allowed in specifications
```

[the remark this answers is not in the document]
Comment (0xaptosj) [reply bulk_migrate/sources/migrate.move]: fixed by using borrow

File: bulk_migrate/Move.toml (hunk 3 above)
[the remark this answers is not in the document]
Comment (0xaptosj) [reply bulk_migrate/Move.toml:5]: put `bulk_migrate = "_"` , follow core and core_v2

**Review by angieyth — APPROVED** [review]

**Pull request merged**